sycl/doc/developer/Debugging.md

```diff
@@ -0,0 +1,167 @@
+# Debugging DPC++
+
+## Building DPC++ in debug mode
+
+To build DPC++ in debug mode you can simply pass `-t Debug` to `configure.py`,
+however debug builds can be quite large and slow so using the following `CMake`
+options may help:
+
+- `-DLLVM_USE_SPLIT_DWARF=ON`: Use `-gsplit-dwarf`, this splits some of the
+  debug information out of the object files into their own separate files,
+  which reduces the size of the object files the linker has to load (see
+  [DebugFission](https://gcc.gnu.org/wiki/DebugFission)).
+- `-DLLVM_PARALLEL_LINK_JOBS=4`: Reduce the number of link jobs running in
+  parallel, to avoid running out of RAM when linking large debug build objects.
+- `-DLLVM_USE_LINKER=lld`: Use the LLVM linker `lld` instead of the GNU linker
+  `ld` as `lld` is usually faster.
+- `-DLIBCLC_CUSTOM_LLVM_TOOLS_BINARY_DIR=<path/to/release/build/bin>`: Use a
+  separate release build of DPC++ as the compiler for building the bitcode
+  libraries (`libclc` and `libdevice`), it normally uses the compiler from the
+  same build, but debug clang is quite slow so using a separate release clang
+  can be worth it.
+
+## Tracing the SYCL runtime
+
+The SYCL runtime is built on top of the Unified Runtime API, using the following
+environment variable prints out all the calls to the Unified Runtime emitted by
+the SYCL runtime, which can help understand runtime behavior:
+
+- `SYCL_UR_TRACE=`: Enables SYCL runtime tracing
+  - `1` for basic tracing, `2` for UR call tracing, `-1` for everything.
+
+## Debugging the compiler
+
+### Clang Driver
+
+- `-###`: Prints each command emitted by the clang driver during compilation.
+  - Can be used to manually replay a compilation command step-by-step to narrow
+    down where a crash happened.
+
+### Middle-end and back-end
+
+LLVM has a number of ways to debug LLVM IR passes and lower, the following
+options illustrate a few of them:
+
+- `-save-temps`: Dump all compilation intermediary files.
```

Comment:
I would also mention `-v` here, because even if all temporary files are saved you won't know their names.